lib/logstash/codecs/identity_map_codec.rb

```diff
@@ -0,0 +1,176 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "concurrent"
+
+# This class is a Codec duck type
+# Using Composition, it maps from a stream identity to
+# a cloned codec instance via the same API as a Codec
+
+module LogStash module Codecs class IdentityMapCodec
+  # subclass of Exception, LS has more than limit (100) active streams
+  class IdentityMapUpperLimitException < Exception; end
```

Comment:
unless there is a specific reason for it, error/exception classes should preferably descend from `StandardError` because when descending from `Exception` a bare rescue clause will not catch them.